ote/src/cljs/ote/ui/form_fields.cljs

```diff
@@ -777,7 +778,72 @@
+            (let [{:keys [errors missing-required-fields]} (and error-data
+                                                                (< i (count error-data))
+                                                                (nth error-data i))]
+              ^{:key (str "div-" i)}
+              [:div
+               [:div.row
+                (doall
+                  (for [{:keys [name label read write type component div-style] :as div-component} table-fields
+                        :let [value ((or read name) row)
+                              field-error (get errors name)
+                              missing? (get missing-required-fields name)
+                              update-fn (if write
+                                          #(update data i write %)
+                                          #(assoc-in data [i name] %))]]
+                    ^{:key (str name label type)}
+
+                    [:div {:class div-class
+                           :style (merge
+                                    style-form/div-form-field
+                                    div-style)}
+                     (if (= :component type)
+                       ;; Render component
+                       (component {:update-form! #(update! (update-fn %))
+                                   :div-table? true
+                                   :row-number i
+                                   :data value})
+                       ;; Render field
+                       [field (merge
+                                (assoc div-component :update! #(update! (update-fn %))
+                                                     :div-table? true)
+                                (when missing?
+                                  {:warning (tr [:common-texts :required-field])})
+                                (when field-error
+                                  {:error field-error}))
+                        value])]))]
+               (when delete?
+                 [:div.row
+                  [:div {:class div-class
+                         :style style-form-fields/table-row-column}
+                   [buttons/delete-table-row {:on-click #(update! (vec (concat (when (pos? i)
+                                                                                 (take i data))
+                                                                               (drop (inc i) data))))}
+                    delete-label]]])]))
+          data))]
+      (when add-label
+       [:div (stylefy/use-style style-base/button-add-row)
+        [buttons/save (merge {:on-click #(update! (conj (or data []) {}))
+                              :disabled (if add-label-disabled?
+                                          (add-label-disabled? (last data))
+                                          (values/effectively-empty? (last data)))}
+                             (when (not (nil? id))
+                               {:id (str id "-button")})) add-label]])]))
```

Comment:
id will be "tableid-button", should id be more specific in order to avoid any conflicts in future? Though today there shouldn't be any :)
```suggestion
                               {:id (str id "-button-save")})) add-label]])]))
```